optout/s3_file_handler.go

```diff
@@ -1 +1,220 @@
+			// Skip files with a bad name.  An unknown file in this dir isn't a blocker
+			handler.Warningf("Unknown file found: %s. Skipping.", metadata)
+			skipped = skipped + 1
+			continue
+		}
+
+		result = append(result, &metadata)
+	}
+
+	return &result, skipped, err
+}
+
+func (handler *S3FileHandler) OpenFile(metadata *OptOutFilenameMetadata) (*bufio.Scanner, func(), error) {
+	handler.Infof("Opening file %s", metadata.FilePath)
+	bucket, file, err := parseS3Uri(metadata.FilePath)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	sess, err := handler.createSession()
+	if err != nil {
+		return nil, nil, err
+	}
+
+	downloader := s3manager.NewDownloader(sess)
+	buff := &aws.WriteAtBuffer{}
+	numBytes, err := downloader.Download(buff, &s3.GetObjectInput{
+		Bucket: aws.String(bucket),
+		Key:    aws.String(file),
+	})
+
+	if err != nil {
+		handler.Errorf("Failed to download bucket %s, key %s", bucket, file)
+		return nil, nil, err
+	}
+
+	handler.Infof("file downloaded: size=%d", numBytes)
+
+	byte_arr := buff.Bytes()
+	sc := bufio.NewScanner(bytes.NewReader(byte_arr))
+	return sc, func() {}, err
+}
+
+func (handler *S3FileHandler) CleanupOptOutFiles(suppresslist []*OptOutFilenameMetadata) error {
+	sess, err := handler.createSession()
+	if err != nil {
+		return err
+	}
+
+	errCount := 0
+	for _, suppressionFile := range suppresslist {
+		if !suppressionFile.Imported {
+			// Don't do anything. The S3 bucket should have a retention policy that
+			// automatically cleans up files after a specified period of time,
+			handler.Warningf("File %s was not imported successfully. Skipping cleanup.", suppressionFile)
+			continue
+		}
+
+		handler.Infof("Cleaning up file %s", suppressionFile)
+
+		bucket, file, err := parseS3Uri(suppressionFile.FilePath)
+		if err != nil {
+			return err
+		}
+
+		svc := s3.New(sess)
+		_, err = svc.DeleteObject(&s3.DeleteObjectInput{Bucket: aws.String(bucket), Key: aws.String(file)})
+
+		if err != nil {
+			handler.Errorf("File %s failed to clean up properly, error occurred while deleting object: %v", suppressionFile, err)
+			errCount++
+			continue
+		}
+
+		err = svc.WaitUntilObjectNotExists(&s3.HeadObjectInput{
+			Bucket: aws.String(bucket),
+			Key:    aws.String(file),
+		})
+
+		if err != nil {
+			handler.Errorf("File %s failed to clean up properly, error occurred while waiting for object to be deleted: %v", suppressionFile, err)
+			errCount++
+			continue
+		}
+
+		handler.Infof("File %s successfully ingested and deleted from S3.", suppressionFile)
+	}
+
+	if errCount > 0 {
+		return fmt.Errorf("%d files could not be cleaned up", errCount)
+	}
+
+	return nil
+}
+
+// Creates a new AWS S3 session. If the handler is given a custom S3 endpoint
+// and/or IAM role ARN to assume, the new session connects using those parameters.
+func (handler *S3FileHandler) createSession() (*session.Session, error) {
+	sess := session.Must(session.NewSession())
+
+	config := aws.Config{
+		Region: aws.String("us-east-1"),
```

Comment:
Do we see a scenario where a file may _not_ be in us-east-1?

Reply:
Ah thanks, I said on our call earlier that it would automatically use whatever is configured in the AWS profile, but I forgot that this is hardcoded.

The BFD S3 buckets will live in us-east-1 so I think we can leave them hardcoded for now to avoid the "too many options/parameters/environment variables" fatigue, but open to alternative thoughts! 

I did note that this region is hardcoded in several other places in the codebase (e.g. the insights pipeline and a looot of Jenkins workflows)